samples/matter/lock/src/app_task.cpp

```diff
@@ -78,54 +57,19 @@ constexpr uint32_t kSwitchTransportTimeout = 10000;
 Identify sIdentify = { kLockEndpointId, AppTask::IdentifyStartHandler, AppTask::IdentifyStopHandler,
 		       Clusters::Identify::IdentifyTypeEnum::kVisibleIndicator };
 
-#ifdef CONFIG_CHIP_WIFI
-app::Clusters::NetworkCommissioning::Instance
-	sWiFiCommissioningInstance(0, &(NetworkCommissioning::NrfWiFiDriver::Instance()));
-#endif
-
 CHIP_ERROR AppTask::Init()
 {
 	/* Initialize CHIP stack */
-	LOG_INF("Init CHIP stack");
-
-	CHIP_ERROR err = chip::Platform::MemoryInit();
-	if (err != CHIP_NO_ERROR) {
-		LOG_ERR("Platform::MemoryInit() failed");
-		return err;
-	}
-
-	err = PlatformMgr().InitChipStack();
-	if (err != CHIP_NO_ERROR) {
-		LOG_ERR("PlatformMgr().InitChipStack() failed");
-		return err;
-	}
-
-#if defined(CONFIG_NET_L2_OPENTHREAD)
-	err = ThreadStackMgr().InitThreadStack();
-	if (err != CHIP_NO_ERROR) {
-		LOG_ERR("ThreadStackMgr().InitThreadStack() failed: %s", ErrorStr(err));
-		return err;
-	}
-
-#ifdef CONFIG_OPENTHREAD_MTD_SED
-	err = ConnectivityMgr().SetThreadDeviceType(ConnectivityManager::kThreadDeviceType_SleepyEndDevice);
-#else
-	err = ConnectivityMgr().SetThreadDeviceType(ConnectivityManager::kThreadDeviceType_MinimalEndDevice);
-#endif /* CONFIG_OPENTHREAD_MTD_SED */
-	if (err != CHIP_NO_ERROR) {
-		LOG_ERR("ConnectivityMgr().SetThreadDeviceType() failed: %s", ErrorStr(err));
-		return err;
-	}
-#endif /* CONFIG_NET_L2_OPENTHREAD */
-
 #ifdef CONFIG_THREAD_WIFI_SWITCHING
-	err = ThreadWifiSwitch::StartCurrentTransport();
-	if (err != CHIP_NO_ERROR) {
-		LOG_ERR("ThreadWifiSwitch::StartCurrentTransport() failed: %" CHIP_ERROR_FORMAT, err.Format());
-		return err;
-	}
-#elif defined(CONFIG_CHIP_WIFI)
-	sWiFiCommissioningInstance.Init();
+	/* In the Thread/Wi-Fi switchable mode, it is the ThreadWifiSwitch module that owns the NetworkCommissioning
+	   instances, so offload the initialization module from controlling that by explicitly setting the
+	   mNetworkingInstance parameter to nullptr. Otherwise, the initialization module instantiates the
+	   NetworkCommissioning object and handles it by default internally. */
+	ReturnErrorOnFailure(Nordic::Matter::InitChipServer(
+		ChipEventHandler, chip::MakeOptional<Nordic::Matter::ChipInitData>(
```

Comment:
Is there a big value from making this argument optional instead of using a simple pointer with default value set to nullptr? I mean, we still need to do additional check (nullcheck vs Valid() check) and the mapping makes call looking much more complex.

Reply:
I was thinking about it. With the pointer argument we'd need to keep data passed to the method alive all the time. If we want to pass the raw ChipInitData temporary object we'd need to either check validity of all fields in the passed structure or add additional field like `IsDataSet`. But I'll think about it more.

Reply:
Got rid of the Optional.